views/default/output/url.php

```diff
@@ -84,5 +82,29 @@
 unset($vars['is_trusted']);
 unset($vars['confirm']);
 
-$attributes = elgg_format_attributes($vars);
-echo "<a $attributes>$text</a>";
+$vars['class'] = elgg_extract_class($vars, 'elgg-anchor');
+
+$text = elgg_format_element('span', [
+	'class' => 'elgg-anchor-label',
+], $text);
+
+$icon = elgg_extract('icon', $vars, '');
+unset($vars['icon']);
+
+if ($icon && !preg_match('/^</', $icon)) {
```

Comment:
Similarly here. String HTML or array arguments for elgg_view_icon. Let's get some other opinions here.